dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/DubboBootstrap.java

```diff
@@ -831,38 +928,96 @@ private String getRegistryCompatibleAddress(RegistryConfig registryConfig) {
         return metadataAddressBuilder.toString();
     }
 
-    private void loadRemoteConfigs() {
-        // registry ids to registry configs
-        List<RegistryConfig> tmpRegistries = new ArrayList<>();
-        Set<String> registryIds = configManager.getRegistryIds();
-        registryIds.forEach(id -> {
-            if (tmpRegistries.stream().noneMatch(reg -> reg.getId().equals(id))) {
-                tmpRegistries.add(configManager.getRegistry(id).orElseGet(() -> {
-                    RegistryConfig registryConfig = new RegistryConfig();
-                    registryConfig.setId(id);
-                    registryConfig.refresh();
-                    return registryConfig;
-                }));
+    private void loadConfigsFromProps() {
+
+        // application config has load before starting config center
+        // load dubbo.applications.xxx
+        loadConfigs(ApplicationConfig.class);
+
+        // load dubbo.modules.xxx
+        loadConfigs(ModuleConfig.class);
+
+        // load dubbo.monitors.xxx
+        loadConfigs(MonitorConfig.class);
+
+        // load dubbo.metricses.xxx
+        loadConfigs(MetricsConfig.class);
+
+        // load multiple config types:
+        // load dubbo.protocols.xxx
+        loadConfigs(ProtocolConfig.class);
+
+        // load dubbo.registries.xxx
+        loadConfigs(RegistryConfig.class);
+
+        // load dubbo.providers.xxx
+        loadConfigs(ProviderConfig.class);
+
+        // load dubbo.consumers.xxx
+        loadConfigs(ConsumerConfig.class);
+
+        // load dubbo.metadata-report.xxx
+        loadConfigs(MetadataReportConfig.class);
+
+        // config centers has bean loaded before starting config center
+        //loadConfigs(ConfigCenterConfig.class);
+
+    }
+
+    private <T extends AbstractConfig> void loadConfigs(Class<T> cls) {
+        // load multiple configs with id
+        Set<String> configIds = configManager.getConfigIds(cls);
```

Comment:
Where does getConfigId initially come from?
Will there be a case that there're no config ids in ConfigManager but only items in properties file?

```properties
dubbo.registries.id1.address=xxx
dubbo.registries.id2.address=yyy
```

Reply:
`getConfigIds`  will search from `Environment`, find out property keys whose start with the specify prefix, and extract sub ids.

```
    public Set<String> getConfigIds(Class<? extends AbstractConfig> clazz) {
        String prefix = CommonConstants.DUBBO + "." + AbstractConfig.getPluralTagName(clazz) + ".";
        Environment environment = ApplicationModel.getEnvironment();
        return ConfigurationUtils.getSubIds(environment.getConfigurationMaps(), prefix);
    }
```

Reply:
The `getConfigIds` method will be moved to `DubboBootstrap`, which is closer to the `loadConfigs` method.